src/modules/builderView/configuration/AddImageStep.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+
+import {
+  Box,
+  Button,
+  InputLabel,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+
+import { NameTheFrameSettingsNames } from '@/@types';
+import { useNameFrameTranslation } from '@/config/i18n';
+import { hooks, mutations } from '@/config/queryClient';
+import { DESCRIPTION_INPUT_ID } from '@/config/selectors';
+import { NAME_THE_FRAME } from '@/langs/constants';
+import ImageDisplay from '@/modules/common/ImageDisplay';
+import UploadImage from '@/modules/common/UploadImage';
+
+type Props = {
+  moveToNextStep: () => void;
+};
+const AddImageStep = ({ moveToNextStep }: Props): JSX.Element => {
+  const { t } = useNameFrameTranslation();
+  const { data: appSettings } = hooks.useAppSettings();
+  const { mutate: postSetting } = mutations.usePostAppSetting();
+  const { mutate: patchSetting } = mutations.usePatchAppSetting();
+
+  const data = appSettings?.find(
+    ({ name }) => name === NameTheFrameSettingsNames.SettingsData,
+  );
+  const image = appSettings?.find(
+    ({ name }) => name === NameTheFrameSettingsNames.File,
+  );
```

Comment:
Since some time we have the possibility to get settings per name https://github.com/graasp/graasp-apps-query-client/blob/beaf4f6ae730fa09134332c40212d8a31fa440d0/src/hooks/appSetting.ts#L22 I think you can even set the type.